cluster-autoscaler/core/static_autoscaler.go

```diff
@@ -287,6 +287,10 @@ func (a *StaticAutoscaler) RunOnce(currentTime time.Time) errors.AutoscalerError
 		if !calculateUnneededOnly {
 			glog.V(4).Infof("Starting scale down")
 
+			if a.AutoscalingContext.AutoscalingOptions.NodeAutoprovisioningEnabled {
```

Comment:
I'm not sure if this should be behind !calculateUnneededOnly check. How expensive is adding/deleting a nodegroup?

Reply:
This is not about expensive operations but about not deleting node pools too early or under conditions that we would not even allow to delete a node. I added a comment to explain it better.